src/allowlist/AllowList.sol

```diff
@@ -63,6 +72,53 @@ contract AllowList is IAllowListV2, Ownable2StepUpgradeable {
         emit FundPermissionSet(entityId, fundSymbol, isAllowed);
     }
 
+    function _setContractAllowedForFundInternal(address addr, string calldata fundSymbol, bool isAllowed) internal {
+        bool currentValue = contractPermissions[addr][fundSymbol];
```

Comment:
nit: could do "does address have entity id" check in here instead of duplicating in external functions? could even reuse `isAddressAllowedForFund` here for idemptotency check